apps/ledger-live-desktop/src/renderer/components/DeviceAction/index.tsx

```diff
@@ -340,8 +340,11 @@ export const DeviceActionDefaultRendering = <R, H extends States, P>({
     }
   }
 
-  if (device?.modelId === "nanoS" && (request as SwapRequest)?.provider === "thorswap") {
-    return renderHardwareUpdate();
+  if (
```

Comment:
You want to say the swap request provider is `thorswap` or `uniswap`  for a NanoS device ?

Reply:
Yes, if the user is using a nano S and wants to swap with uniswap or Thorswap

Reply:
Then I think you need to update the conditions, the actual code will let all swap requests using `uniswap` pass no matter on which device.

Reply:
` (device?.modelId === "nanoS" && (request as SwapRequest)?.provider === "thorswap") ||
    (request as SwapRequest)?.provider === "uniswap"
  )`
  
  If it's Thorswap or Uniswap and you are on a Nano S you are returned a error drawer, I don't understand

Reply:
If the device is not NanoS, but the swap provider is `uniswap`, the code in the if-scope will still executed.